### DISCUSSION TITLE
Frequency of villages and structures far too high

### DISCUSSION THREAD

**Discussion by TFWIsaac** [AllTheMods/ATM-10#2760] [Modpack Content]
Today I generated several worlds with different seeds using server version 1.7. I noticed that a village or other structures are generated approximately every 150 to 200 blocks. 
In my opinion, this is far too much. Finding a village should be rare to very rare.

On the technical side: I deleted the world folder, set a new seed and then restarted the server.

**Comment by liquidthex** [comment]
I think this is the case, as well.
Also: Biome sizes are INCREDIBLY tiny like if you have a 16 chunk render distance and you fly up you can see no less than 20 different biomes.
Try to find an expansive desert, an expansive plains--they don't exist, every single biome is TINY.

**Reply by liquidthex** [reply]
Unfortunately this is what ended up killing the modpack for me. 
The world is just too messy, you never get to enjoy any part of it because the biome and structure generation are set incredibly poorly.

**Comment by saren42** [comment]
I want to concur on this, this is still a huge issue, you can't swing a dead creeper without hitting 3 structures. This needs to be dramatically decreased. Or figure out this to be a configurable option in world gen for SP or servers possibly. This is still fairly frequent issue in recent updates. Unless this is still waiting to be pushed to a release later on, based on further internal testing that is.

**Comment by mopsyd** [comment]
I honestly can't find a single larger structure that isn't corrupted by usually several other smaller structures spawning inside of it. This is annoying but not a huge deal when it's something like a nether fortress or mineshaft, but it's immensely aggravating when it's a very rare structure needed for progression like a cataclysm dungeon or the stronghold. Villages, meteors, and smaller villager houses are the usual culprits, but pillager towers/forts break a lot too because they are way too prevalent. All of the structures need their frequency tuned significantly on a case by case, because there are some that should be pretty frequent, just not smashed together to a point that they make each other unplayable.

**Reply by Drarthe** [reply]
Once I added Sparse Structures to ATM 10 it made this stop happening on world gen.  I wanted it so I could build large Minecolonies cities, but it's also nice to have huge dungeons that generate properly.

**Comment by pietro-lopes** [comment]
We are well aware of this issue, as some pointed it is not an easy fix.

**Comment by Drarthe** [comment]
I just added Sparse Structures to my atm10 install and it's perfect for me.  Wide open spaces, making structures special when found.
